nixos/modules/services/video/broadcast-box.nix

```diff
@@ -0,0 +1,189 @@
+{ lib, pkgs, config, ... }:
+let
+  inherit (lib)
+    mkIf
+    mdDoc
+    mkEnableOption
+    getExe
+    boolToString
+    types
+    concatStringsSep
+    mkOption
+    optional
+    mkPackageOption;
+
+  cfg = config.services.broadcast-box;
+in
+{
+  config = mkIf cfg.enable {
+    systemd.services.broadcast-box = {
+      description = "Broadcast Box";
+      after = [ "network-online.target" ];
+      wants = [ "network-online.target" ];
+      wantedBy = optional cfg.autoStart "multi-user.target";
+
+      environment = with cfg; {
+        REACT_APP_API_PATH = "/api";
+
+        HTTP_ADDRESS = "${listenAddress}:${toString tcp.port}";
+        UDP_MUX_PORT = mkIf (udp.port != null) (toString udp.port);
+        INTERFACE_FILTER = udp.interfaceFilter;
+
+        NETWORK_TEST_ON_START = boolToString cfg.networkTestOnStart;
+        STUN_SERVERS = concatStringsSep "|" cfg.stunServers;
+
+        ENABLE_HTTP_REDIRECT = mkIf ssl.redirect "true";
+        SSL_CERT = mkIf (ssl.cert != null) (toString ssl.cert);
+        SSL_KEY = mkIf (ssl.key != null) (toString ssl.key);
+
+        NAT_1_TO_1_IP = nat1To1Ip;
+        INCLUDE_PUBLIC_IP_IN_NAT_1_TO_1_IP = mkIf includePublicIpInNat1To1Ip "true";
+      } // extraEnv;
+
+      serviceConfig = {
+        DynamicUser = true;
+        ExecStart = "${getExe cfg.package}";
+        Restart = "always";
+        RestartSec = "10s";
+      };
+    };
+
+    networking.firewall = mkIf cfg.openFirewall {
+      allowedTCPPorts = [ cfg.tcp.port ] ++ optional cfg.ssl.redirect 80;
+      allowedUDPPorts = optional (cfg.udp.port != null) cfg.udp.port;
+    };
+  };
+
+  options.services.broadcast-box = {
+    enable = mkEnableOption "broadcast box";
+    package = mkPackageOption pkgs "broadcast-box" { };
+
+    listenAddress = mkOption {
+      type = types.str;
+      default = "";
+      example = "127.0.0.1";
+      description = mdDoc ''
+        Address the HTTP server will listen on.
+      '';
+    };
+
+    tcp.port = mkOption {
+      type = types.port;
+      default = 8080;
+      description = mdDoc ''
+        TCP Port the HTTP server will listen on.
+      '';
+    };
```

Comment:
Grouping with `http` is a good idea, I'll do that. If I understand the docs correctly, Broadcast Box actually can send WebRTC traffic over TCP with the `TCP_MUX_ADDRESS` variable (which I will add as an option). What do you think about something like:
```
http.listenAddress
http.port

udpMux.port
tcpMux.listenAddress
tcpMux.port
```
That way it clearly differentiates the http vs WebRTC addresses and ports. This would also allow me to add options for `UDP_MUX_PORT_(WHEP|WHIP)` as `udpMux.whep.port` and `udpMux.whip.port`.